src/inputs/EditableLabel/EditableLabel.js

```diff
@@ -0,0 +1,194 @@
+    height: '20px',
+    color: theme.palette.primary.main,
+  },
+  circularLoading: {
+    marginLeft: '15px',
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.background.card,
+  },
+}));
+
+export const EditableLabel = (props) => {
+  const { value, onNewValue, checkValue, labels, ...typographyprops } = props;
+
+  const classes = useStyles();
+
+  const [isEditing, setEditing] = useState(false);
+  const [error, setError] = useState(null);
+  const [isHovered, setHovered] = useState(false);
+  const [textFieldValue, setTextFieldValue] = useState(value);
+
+  const startEdition = (event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setEditing(true);
+  };
+
+  const checkNewValueOnChange = (event) => {
+    const newValue = event.target.value;
+    if (newValue !== value) {
+      setError(checkValue(newValue));
+    }
+  };
+
+  const stopEdition = (event) => {
+    event.stopPropagation();
+    const newValue = event.target.value;
+    if (newValue !== value) {
+      if (error == null) {
+        onNewValue(newValue);
+        setTextFieldValue(newValue);
+      }
+    }
+    setEditing(false);
+    setHovered(false);
+  };
+
+  const getLabelDisplay = () => {
+    if (value === textFieldValue) {
+      return (
+        <div
+          className={classes.labelDisplayContainer}
+          onMouseLeave={() => setHovered(false)}
+          onMouseOver={() => setHovered(true)}
+          onClick={startEdition}
+        >
+          {isHovered && <EditIcon className={classes.editIcon} />}
+          <Typography className={classes.label} variant={typographyprops.variant}>
+            {value}
+          </Typography>
+        </div>
+      );
+    } else {
+      return <CircularProgress size={25} className={classes.circularLoading} />;
+    }
+  };
+
+  return (
+    <div data-cy="editable-label" id="EditableLabel">
+      {isEditing ? (
+        <TextField
+          className={classes.textField}
+          variant="outlined"
+          autoFocus
+          defaultValue={value}
+          error={error != null}
+          helperText={error}
+          placeholder={labels.title}
+          onChange={checkNewValueOnChange}
+          onBlur={(event) => {
+            stopEdition(event);
+          }}
+          onClick={(event) => event.stopPropagation()}
+          onKeyPress={(event) => {
+            if (event.key === 'Enter') {
+              stopEdition(event);
+            }
+          }}
+          onKeyDown={(event) => {
+            if (event.key === 'Escape') {
+              setEditing(false);
+            }
+          }}
+          onFocus={(event) => {
+            event.stopPropagation();
+          }}
+        />
+      ) : (
+        getLabelDisplay()
+      )}
+    </div>
+  );
+};
+
+EditableLabel.propTypes = {
+  /**
+   * String value displayed in label
+   */
+  value: PropTypes.string.isRequired,
+  /**
+   * Function to handle onNewValue event
```

Comment:
This comment could be improved: "_Callback function called when a new valid value is set._"